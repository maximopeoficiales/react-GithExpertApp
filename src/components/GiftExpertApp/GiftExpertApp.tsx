import React, { useState } from "react";
import AddCategory from "../AddCategory/AddCategory";
import GiftGrid from "../GiftGrid/GiftGrid";

interface MyProps {
  categorysStatic: string[];
}
const defaultProps: MyProps = {
  categorysStatic: ["rick and morty"],
};
const GiftExpertApp = (props: MyProps) => {
  props = { ...defaultProps, ...props };

  const [categorys, setCategorys] = useState(props.categorysStatic);
  const handlerClick = (category: string) => {
    setCategorys([category, ...categorys]);
  };
  return (
    <div>
      <div className="animate__animated animate__bounce">
        <h2 className="title-app">GiftAppReact</h2>
        <AddCategory addCategory={handlerClick} />
      </div>

      <div className="">
        {categorys.map((e) => {
          return <GiftGrid key={e} category={e} />;
        })}
      </div>
    </div>
  );
};

export default GiftExpertApp;
