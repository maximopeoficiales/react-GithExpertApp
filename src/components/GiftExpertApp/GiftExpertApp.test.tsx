import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import GiftExpertApp from "./GiftExpertApp";
import { shallow } from "enzyme";

describe("<GiftExpertApp />", () => {
  const categorys = ["rick y morty"];
  test("it should mount", () => {
    const wrapper = shallow(<GiftExpertApp categorysStatic={categorys} />);
    // verifica si se renderizo correctamente
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GiftGrid").length).toBe(categorys.length);
  });
});
