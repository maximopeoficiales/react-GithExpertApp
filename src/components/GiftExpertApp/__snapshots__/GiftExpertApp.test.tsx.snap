// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftExpertApp /> it should mount 1`] = `
<div>
  <div
    className="animate__animated animate__bounce"
  >
    <h2
      className="title-app"
    >
      GiftAppReact
    </h2>
    <AddCategory
      addCategory={[Function]}
    />
  </div>
  <div
    className=""
  >
    <GiftGrid
      category="rick y morty"
      key="rick y morty"
    />
  </div>
</div>
`;
